js/bybit.js

```diff
@@ -3277,126 +4321,307 @@ module.exports = class bybit extends Exchange {
         }
         const exchangeSpecificParam = this.safeString (params, 'time_in_force');
         const timeInForce = this.safeStringLower (params, 'timeInForce');
         const postOnly = this.isPostOnly (isMarket, exchangeSpecificParam === 'PostOnly', params);
         if (postOnly) {
-            request['time_in_force'] = 'PostOnly';
+            request['timeInForce'] = 'PostOnly';
         } else if (timeInForce === 'gtc') {
-            request['time_in_force'] = 'GoodTillCancel';
+            request['timeInForce'] = 'GoodTillCancel';
         } else if (timeInForce === 'fok') {
-            request['time_in_force'] = 'FillOrKill';
+            request['timeInForce'] = 'FillOrKill';
         } else if (timeInForce === 'ioc') {
-            request['time_in_force'] = 'ImmediateOrCancel';
+            request['timeInForce'] = 'ImmediateOrCancel';
         }
-        const triggerPrice = this.safeValueN (params, [ 'stopPrice', 'triggerPrice', 'stop_px' ]);
-        const isTriggerOrder = triggerPrice !== undefined;
-        const stopLossPrice = this.safeValue (params, 'stopLossPrice');
-        const isStopLossOrder = stopLossPrice !== undefined;
-        const takeProfitPrice = this.safeValue (params, 'takeProfitPrice');
-        const isTakeProfitOrder = takeProfitPrice !== undefined;
-        if (isTriggerOrder) {
-            request['trigger_by'] = 'LastPrice';
-            const preciseStopPrice = this.priceToPrecision (symbol, triggerPrice);
-            request['stop_px'] = parseFloat (preciseStopPrice);
-            const basePrice = this.safeValue2 (params, 'base_price', 'basePrice');
-            if (basePrice === undefined) {
-                throw new ArgumentsRequired (this.id + ' createOrder() requires a base_price parameter for trigger orders, your triggerPrice > max(market price, base_price) or triggerPrice < min(market price, base_price)');
+        if (market['swap']) {
+            const triggerPrice = this.safeValue2 (params, 'stopPrice', 'triggerPrice');
+            const stopLossPrice = this.safeValue (params, 'stopLossPrice', triggerPrice);
+            const isStopLossOrder = stopLossPrice !== undefined;
+            const takeProfitPrice = this.safeValue (params, 'takeProfitPrice');
+            const isTakeProfitOrder = takeProfitPrice !== undefined;
+            const isStopOrder = isStopLossOrder || isTakeProfitOrder;
+            if (isStopOrder) {
+                request['triggerBy'] = 'LastPrice';
+                const stopPx = isStopLossOrder ? stopLossPrice : takeProfitPrice;
+                const preciseStopPrice = this.priceToPrecision (symbol, stopPx);
+                request['triggerPrice'] = preciseStopPrice;
+                const delta = this.numberToString (market['precision']['price']);
+                request['basePrice'] = isStopLossOrder ? Precise.stringSub (preciseStopPrice, delta) : Precise.stringAdd (preciseStopPrice, delta);
```

Comment:
Hum I think `basePrice` does not exist anymore, now they have `triggerDirection` 
https://bybit-exchange.github.io/docs/derivativesV3/contract/#t-dv_placeorder

Reply:
I think it's gonna be:

```JS
request['triggerDirection'] = isStopLossOrder ? 2 : 1; // 1: rise; 2: fall
```